elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -174,57 +175,114 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
         }
     }
 
+    @SuppressWarnings("unchecked")
     private void populateInheritance(ElideTableConfig elideTableConfig) {
-        for (Table table : elideTableConfig.getTables()) {
+        //ensures validation is run before populate always.
+        validateInheritance(this.elideTableConfig);
+
+        List<Table> tables = elideTableConfig.getTables().stream().collect(Collectors.toList());
+        while (!tables.isEmpty()) {
+            Table table = tables.remove(0);
             if (table.hasParent()) {
-                Map<String, Measure> measures = getInheritedMeasures(table, new HashMap<String, Measure>());
```

Comment:
We should leave the methods getInheritedMeasures and getInheritedDimensions, etc.  They improve readability.  Those functions can return the correct type and in turn call getInheritedAttribute with the appropriate Inherit method.